src/util_misc.cpp

```diff
@@ -274,23 +274,40 @@ void Util::dupFds(int oldfd, const vector<int>& newfds)
 
 
 /* Begin miscellaneous/helper functions. */
-// Reads from fd until count bytes are read, or newline encountered.
-// Returns NULL at EOF.
-// FIXME: count is unused. Buffer-overrun possible
+
+/* Reads from fd until count bytes are read, or
+ * newline encountered.
+ *
+ * Side effects: Copies the characters, including
+ * the newline, read from the fd into the buf.
+ *
+ * Returns num of chars read on success;
+ *         -1 on read failure or invalid args; and
+ *         -2 if the buffer is too small
+ */
 int Util::readLine(int fd, char *buf, int count)
 {
   int i = 0;
   char c;
+  if (fd < 0 || buf == NULL) return -1;
```

Comment:
It's a nice catch to guard against `fd < 0`!  However, I wonder if we should do a JASSERT in this case, instead of leaving it up to the caller of `Util::readLine()` to check the return value.  If `fd < 0`, then the programmer should have checked for this _before_ calling `Util::readLine()`, rather than looking at the return value, and trying to fix it after the fact.